### PR TITLE
Feature/config check socket file

### PR DESCRIPTION
**Pull request by tty2**
Instead of check if u.Path is not empty only, check if file we want too use in unix socket connection exists.
Fail fast, before trying to ping.

### CODE REVIEW

**Comment by vmihailenco** [comment]
I wonder why "Fail fast, before trying to ping"? This function parses the config - it is surprising that it also tried to connect / check if file exists etc.

**Comment by tty2** [comment]
Yes, It's hard to follow the thoughts I want to say. 

I mean we call first `ParseURL, NewClient` and then `Ping`, and we fail before calling `NewClient` and `Ping` if file does not exist.
Sorry for the confusing message.

**Comment by vmihailenco** [comment]
Closing since `ParseURL` should not be concerned whether socket exists or not. You still can `ParseURL` and check socket with `os.Stat` before calling `NewClient` - if you need it.

**Comment by tty2** [comment]
> You still can `ParseURL` and check socket with `os.Stat` before calling `NewClient` - if you need it.

It makes sense. I guess you are right.
